contribute/views.py

```diff
@@ -1,9 +1,34 @@
+
+
 from django.views.generic import TemplateView
 
+from user.models import Profile, UserLanguage
+from word.models import Tag, Word, WordVersion
+
 
 class ContribView(TemplateView):
     template_name = 'contribute/main.html'
 
+    def get(self, request, *args, **kwargs):
+        context = self.get_context_data(**kwargs)
+
+        try:
+            user_profile = Profile.objects.get(user=request.user)
+        except TypeError:
+            user_profile = None
+
+        if user_profile:
```

Comment:
Instead of using another if, you could use the "else" clause for the "try/except". Then you wouldn't need to set the "user_profile = None" in the Exception.

Reply:
thanks